src/main/java/seedu/address/model/property/PropertyAddressPredicate.java

```diff
@@ -13,8 +13,11 @@
     /**
      * Constructs a PropertyAddressPredicate.
      */
-    public PropertyAddressPredicate(String address) {
+    public PropertyAddressPredicate(String address) throws IllegalArgumentException {
         requireNonNull(address);
+        if (address.trim().isEmpty()) {
+            throw new IllegalArgumentException("Address cannot be empty. ");
```

Comment:
Maybe change to `"Address given is empty"` to make error message consistent throughout?